libs/kotaemon/kotaemon/indices/ingests/extensions.py

```diff
@@ -0,0 +1,140 @@
+from copy import deepcopy
+from typing import Any
+
+from decouple import config
+from llama_index.core.readers.base import BaseReader
+from theflow.settings import settings as flowsettings
+
+from kotaemon.loaders import (
+    AdobeReader,
+    AzureAIDocumentIntelligenceLoader,
+    GOCR2ImageReader,

```

Comment:
**Suggestion:** Add error handling during the initialization of `GOCR2ImageReader`.

**Existing Code:**
```Python
GOCR2ImageReader,
```

**Improved Code:**
```Python
try:
    self.gocr2_reader = GOCR2ImageReader()
except Exception as e:
    raise RuntimeError(f"Failed to initialize GOCR2ImageReader: {e}")
```

**Details:**
Ensure that the `GOCR2ImageReader` is properly initialized with error handling to manage potential issues during instantiation.

Reply:
In the current implementation and your suggestion. Both raise error when initialization failed. What's the difference here ?